### ISSUE TITLE
From CreateToPointCloud2Msg to sensor_msgs.PointCloud2

### ISSUE THREAD

**Issue by tepsaj** [neka-nat/cupoch#76]
Hi @neka-nat, 

I'm having problems with integrating the output of the `CreateToPointCloud2Msg` -function to ROS sensor_msgs PointCloud2 message. I was wondering if you could help? 

Here is the code that I use in my ROS node callback to create and publish PointCloud2 message.
```python
import cupoch as cph
import numpy as np
from sensor_msgs.msg import PointCloud2
pub = rospy.Publisher('filtered_pcloud', PointCloud2, queue_size = 1)

# pcd is my point cloud.
temp_msg = cph.io.create_to_pointcloud2_msg(pcd)

msg = PointCloud2()
msg.width = temp_msg[1].width 
msg.height = temp_msg[1].height 
msg.fields = temp_msg[1].fields
msg.is_bigendian = temp_msg[1].is_bigendian
msg.point_step = temp_msg[1].point_step
msg.is_dense = temp_msg[1].is_dense
msg.row_step = temp_msg[1].row_step

msg.data = temp_msg[0]
pub.publish(msg)
```

Then I subscribe to the topic with MATLAB over network and save the messages. The problem is that the data is having strange transformation along the way. The first three points of my point cloud are as follow, printed with command inside the callback
```python
points = np.asarray(pcd.points.cpu())
print(points[0])
print(points[1])
print(points[2])
```

> `[-11.926463 -4.7370405 19.449484]`
> `[-11.026707 -5.486838 19.449484]`
> `[-10.876322 -4.323442 17.68135]`

Then here is how the data looks in MATLAB, which I believe is uint8? This is straight from the variable explorer in MATLAB.

![matlab](https://i.imgur.com/00YWxr5.png)

In MATLAB I use the rosReadXYZ() function from MATLAB ROS Toolbox to transform the points to single format (same as C++ float). Now the points don't correspond at all to the original point cloud.

> `1.3354122e+19	9.5873945e+11	0.00020158001`
> `3.0757667e-06	7.4418314e+28	1.0574327e-05`
> `2.9203201e+32	7.5634546e+28	8.6881560e+11`

Here is also print of the points inside the pointcloud2 message before publishing it.
> `b'_S9_S9_S9_S9_S9_S9_E'`

which is printed with this code
```python
print(temp_msg[0][0:20])
```
Thank you in advance!

-tepsaj

**Comment by mvanlobensels** [comment]
Hi @neka-nat,

I am running into the issue that when I convert a PointCloud2 into a Cupoch pointcloud, and then back again to PointCloud2 without any filtering, I do not get the original point cloud back. Is this issue related?

This is the original:
![image](https://user-images.githubusercontent.com/50023316/131819260-f067f53d-8df0-4560-a2b5-88305aa31976.png)

This is what I get after conversion:
![image](https://user-images.githubusercontent.com/50023316/131819397-619b839f-c320-46c5-bfdb-be36d9348591.png)

**Comment by neka-nat** [comment]
I released [v0.2.2.0](https://pypi.org/project/cupoch/).
Please try it!
